### PR TITLE
[WIP] Evaluate RAGs by using Hugging face's dataset.

### PR DESCRIPTION
**Pull request by lxngu**

### CODE REVIEW

**Comment by crazycth** [comment]
@lxngu Is this pr still work in progress?

File: demo/data/rags_compare.csv [hunk 1]
```diff
@@ -0,0 +1,10 @@
+question
```
Comment (crazycth) [inline demo/data/rags_compare.csv:1]: Can we automatically generate questions now ?

or we can just upload questions in csv format

File: litellm_uuid.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+bf3b2d92-c935-44ca-8523-e3130033a85c
```
Comment (crazycth) [inline litellm_uuid.txt:1]: Remove this file ?

File: pyproject.toml [hunk 3]
```diff
@@ -15,7 +15,7 @@ packages = [{ include = "yival", from = "src" }]
 yival = 'yival.__main__:main'
 
 [tool.poetry.dependencies]
-python = ">=3.10,<3.13"
+python = ">=3.10,<3.12"
```
Comment (crazycth) [inline pyproject.toml:18]: We're getting low on supported versions of python, is there a way to set both langchain and llama-index to extra?

File: src/yival/schemas/evaluator_config.py [hunk 4]
```diff
@@ -185,3 +185,11 @@ class BertScoreEvaluatorConfig(EvaluatorConfig):
...
     lan: str = 'zh'
     indicator: str = 'p'  # p,r,f
+
+
+@dataclass
+class Rags_EvaluatorConfig(EvaluatorConfig):
+    evaluator_type: EvaluatorType = EvaluatorType.INDIVIDUAL
+    desciption: str = " This is the description of the evaluator"
+
+    metric_type: str = ''
```
Comment (crazycth) [inline src/yival/schemas/evaluator_config.py:195]: add some instruction on how to set metric_type

File: demo/rags_compare.py [hunk 5]
```diff
@@ -0,0 +1,489 @@
...
+        if retriever == 'llamaindex':
+            context = deal_llamaindex(context=context)
+        else:
+            context = deal_context(context=context)
+    r_m_ps.init_retriever(retriever=retriever, context=context)
+
+    if retriever == 'MultiQueryRetriever':
+        flag = 1
+        while r_m_ps.inited[retriever] == False:
+            print('line:302')
```
Comment (crazycth) [inline demo/rags_compare.py:359]: clean debug code here(print XXX) ? or make it more clear

Comment (crazycth) [inline demo/rags_compare.py:359]: clean debug code here(print XXX) ? or make it more clear

File: demo/rags_compare.py [hunk 6]
```diff
@@ -0,0 +1,489 @@
...
+            time.sleep(1)
+        # docs=r_m_ps.retrievers[retriever].vectorstore.similarity_search(input)
+        docs = r_m_ps.retrievers[retriever].get_relevant_documents(query=input)
+
+        res = langchain_docs_to_string(docs=docs)
+        pass
+    elif retriever == 'llamaindex':
+        flag = 1
+        while r_m_ps.inited[retriever] == False:
+            print('line:362')
```
Comment (uni-zhuan) [inline demo/rags_compare.py:425]: also clean up this code

File: demo/data/rags_compare.csv (hunk 1 above)
Comment (lxngu) [reply demo/data/rags_compare.csv:1]: We can only upload questions in csv format now, and the current version is using the questions from huggingface datasets. Uploading our questions need different params for the class ‘retriever_model_prompt' to init.

File: pyproject.toml (hunk 3 above)
Comment (lxngu) [reply pyproject.toml:18]: Yes, I remeber that someone, maybe langchain llama-index or ragas, needs this change.

File: src/yival/schemas/evaluator_config.py (hunk 4 above)
Comment (lxngu) [reply src/yival/schemas/evaluator_config.py:195]: ok

File: demo/rags_compare.py (hunk 5 above)
Comment (lxngu) [reply demo/rags_compare.py:359]: ok

File: demo/rags_compare.py (hunk 6 above)
Comment (lxngu) [reply demo/rags_compare.py:425]: ok

**Pull request closed without merging**